src/osd/OSD.cc

```diff
@@ -591,6 +591,14 @@ void OSDService::activate_map()
   agent_lock.Unlock();
 }
 
+void OSDService::request_osdmap_update(epoch_t e)
+{
+  std::lock_guard<std::mutex> l(osd->need_epoch_lock);
+  if (e > osd->subscribed_epoch) {
+    osd->need_epoch = e;
```

Comment:
request_osdmap() doesn't need osd_lock.. it just grabs the latest osdmap and then calls into monc.  Is there a lock ordering problem if we do that?

Reply:
good point, I was just not looking hard enough through the existing calls which did hold the osd_lock. looks like there's no issue, so I'll do that.